### ISSUE TITLE
Stats Crash: Crash while backgrounding/foregrounding

### ISSUE THREAD

**Issue by kurzee** [wordpress-mobile/WordPress-iOS#6273]
### Steps to reproduce the behavior

1. App was freshly installed, testing out Reader state restoration.
2. Background app in simulator while on Reader tab.
3. Foreground app, crash in Stats.

##### Tested on [device], iOS [version], WPiOS [version]
iPhone 6 Plus simulator, iOS 10.1

```
2016-12-05 15:35:48:411 WordPress[18451:5675674] <WordPressAppDelegate: 0x6000000b31a0> applicationWillResignActive:
2016-12-05 15:35:48:996 WordPress[18451:5675674] <WordPressAppDelegate: 0x6000000b31a0> applicationDidEnterBackground:
2016-12-05 15:35:57:905 WordPress[18451:5675674] <WordPressAppDelegate: 0x6000000b31a0> applicationWillEnterForeground:
2016-12-05 15:35:58:413 WordPress[18451:5675674] <WordPressAppDelegate: 0x6000000b31a0> applicationDidBecomeActive:
2016-12-05 15:35:58.833 WordPress[18451:5675674] *** Assertion failure in -[WPStatsCollectionViewFlowLayout _getSizingInfosWithExistingSizingDictionary:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3600.5.2/UICollectionViewFlowLayout.m:1343
2016-12-05 15:35:58.844 WordPress[18451:5675674] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'negative sizes are not supported in the flow layout'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010da6634b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010d4ce21e objc_exception_throw + 48
	2   CoreFoundation                      0x000000010da6a442 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x0000000107bcbe4d -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 195
	4   UIKit                               0x000000010938f280 -[UICollectionViewFlowLayout _getSizingInfosWithExistingSizingDictionary:] + 3418
	5   UIKit                               0x0000000109390887 -[UICollectionViewFlowLayout _fetchItemsInfoForRect:] + 127
	6   UIKit                               0x0000000109389410 -[UICollectionViewFlowLayout prepareLayout] + 273
	7   UIKit                               0x00000001093a9c78 -[UICollectionViewData _prepareToLoadData] + 159
	8   UIKit                               0x00000001093aa524 -[UICollectionViewData validateLayoutInRect:] + 57
	9   UIKit                               0x00000001093518c2 -[UICollectionView layoutSubviews] + 232
	10  UIKit                               0x0000000108ac8f50 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1237
	11  QuartzCore                          0x000000010859fcc4 -[CALayer layoutSublayers] + 146
	12  QuartzCore                          0x0000000108593788 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 366
	13  UIKit                               0x0000000108ab6ff0 -[UIView(Hierarchy) layoutBelowIfNeeded] + 1507
	14  UIKit                               0x0000000108abd6e6 +[UIView(Animation) performWithoutAnimation:] + 90
	15  UIKit                               0x0000000108b4bad2 __46-[UITableView _updateWithItems:updateSupport:]_block_invoke.996 + 2919
	16  UIKit                               0x0000000108abea6d +[UIView(UIViewAnimationWithBlocks) _setupAnimationWithDuration:delay:view:options:factory:animations:start:animationStateGenerator:completion:] + 581
	17  UIKit                               0x0000000108abef74 +[UIView(UIViewAnimationWithBlocks) animateWithDuration:delay:options:animations:completion:] + 105
	18  UIKit                               0x0000000108b4ac06 -[UITableView _updateWithItems:updateSupport:] + 4426
	19  UIKit                               0x0000000108b42acf -[UITableView _endCellAnimationsWithContext:] + 17960
	20  WordPressComStatsiOS                0x000000010a313ffa __51-[StatsTableViewController retrieveStatsSkipGraph:]_block_invoke.301 + 826
	21  WordPressComStatsiOS                0x000000010a333c2b __85-[WPStatsService remoteItemCompletionWithCache:forStatsSection:andCompletionHandler:]_block_invoke + 443
	22  WordPressComStatsiOS                0x000000010a347177 __88-[WPStatsServiceRemote operationForVideosForDate:andUnit:viewAll:withCompletionHandler:]_block_invoke + 2167
	23  AFNetworking                        0x0000000106ac8c84 __116-[AFHTTPSessionManager dataTaskWithHTTPMethod:URLString:parameters:uploadProgress:downloadProgress:success:failure:]_block_invoke.97 + 228
	24  AFNetworking                        0x0000000106aeb819 __72-[AFURLSessionManagerTaskDelegate URLSession:task:didCompleteWithError:]_block_invoke_2.150 + 201
	25  libdispatch.dylib                   0x000000010e90b980 _dispatch_call_block_and_release + 12
	26  libdispatch.dylib                   0x000000010e9350cd _dispatch_client_callout + 8
	27  libdispatch.dylib                   0x000000010e915a1d _dispatch_main_queue_callback_4CF + 733
	28  CoreFoundation                      0x000000010da2a4f9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	29  CoreFoundation                      0x000000010d9eff8d __CFRunLoopRun + 2205
	30  CoreFoundation                      0x000000010d9ef494 CFRunLoopRunSpecific + 420
	31  GraphicsServices                    0x00000001104c7a6f GSEventRunModal + 161
	32  UIKit                               0x0000000108a04964 UIApplicationMain + 159
	33  WordPress                           0x0000000104cc635e main + 62
	34  libdyld.dylib                       0x000000010e98168d start + 1
	35  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by astralbodies** [comment]
I'm looking into this now - thanks @kurzee!

**Comment by astralbodies** [comment]
Fixed with #6342.
